profiles/minimal/nsswitch.conf

```diff
@@ -1,14 +1,14 @@
-aliases:    files                                       {exclude if "with-custom-aliases"}
-automount:  files                                       {exclude if "with-custom-automount"}
-ethers:     files                                       {exclude if "with-custom-ethers"}
-group:      files {if "with-altfiles":altfiles }systemd {exclude if "with-custom-group"}
-hosts:      files dns myhostname                        {exclude if "with-custom-hosts"}
-initgroups: files                                       {exclude if "with-custom-initgroups"}
-netgroup:   files                                       {exclude if "with-custom-netgroup"}
-networks:   files                                       {exclude if "with-custom-networks"}
-passwd:     files {if "with-altfiles":altfiles }systemd {exclude if "with-custom-passwd"}
-protocols:  files                                       {exclude if "with-custom-protocols"}
-publickey:  files                                       {exclude if "with-custom-publickey"}
-rpc:        files                                       {exclude if "with-custom-rpc"}
-services:   files                                       {exclude if "with-custom-services"}
-shadow:     files                                       {exclude if "with-custom-shadow"}
\ No newline at end of file
+aliases:    files                                          {exclude if "with-custom-aliases"}
+automount:  files                                          {exclude if "with-custom-automount"}
+ethers:     files                                          {exclude if "with-custom-ethers"}
+group:      files {if "with-altfiles":altfiles }systemd    {exclude if "with-custom-group"}
+hosts:      files resolve [!UNAVAIL=return] dns myhostname {exclude if "with-custom-hosts"}
```

Comment:
No, the order should be 
`resolve [!UNAVAIL=return] myhostname files dns`

Justification: `systemd-resolved` caches files internally, so that the file doesn't need to be parsed by each client. Some people put long lists here, e.g. blacklists. So `resolve` should be before `files`. It should also be before `dns`, so that `dns` only works as fallback. And `resolve` implements the same synthetization that `myhostname` does, so `myhostname` should be after `resolve`. And it should be right after `resolve` to minimize the difference in behaviour for synthetized names in case systemd-resolved is active and in case it stopped working. Additionally `myhostname` is more trusted than `dns`, so it should be before it. It is also more up-to-date than `files` most of the time, and should be before it too.

Reply:
OK... sorry @pbrezina for telling you to use the wrong order, that's my bad.

What about mdns4_minimal (which is not part of this authselect config, but which we also have enabled by default)?

Reply:
According to nss-resolve(8), the recommended order is:

`hosts:          files mymachines resolve [!UNAVAIL=return] dns myhostname`

But we don't use mymachines in Fedora due to some bug related to interaction with freeipa, so that should be ignored. The difference from your suggestion is the placement of `files`.

> systemd-resolved caches files internally, so that the file doesn't need to be parsed by each client. Some people put long lists here, e.g. blacklists. So resolve should be before files.

Then I suggest updating the nss-resolve manpage.

Reply:
That's a good question. nss-mdns is enabled by %post scriptlet, and it inserts itself directly before dns. So I think we should keep this order, i.e. have something like
```
resolve [!UNAVAIL=return] myhostname files mdns4_minimal [NOTFOUND=return] dns
```
But maybe this does the wrong thing to do? This means that resolved will be used to resolve mdns names (if it is running).

Reply:
> According to nss-resolve(8), the recommended order is:
> 
> `hosts: files mymachines resolve [!UNAVAIL=return] dns myhostname`

This is the old version. The package in rawhide and https://www.freedesktop.org/software/systemd/man/nss-resolve.html#Example have the updated one.

Reply:
AFAIK resolved also has some support for mdns. Is nss_mdns even needed when resolved is used?

Reply:
In theory, I would think it would not be needed. I think Zbigniew wants to keep it around anyway though, "just in case"...? After all, resolved can be disabled without editing nsswitch.conf.

Reply:
Yeah, the assumption is that people might want to disable `systemd-resolved.service` and things should continue working as much as possible in that case, falling back to existing mechanisms for name resolution.

Reply:
Ok, so I will keep it without nss_mdns for now as there was no request to add it to these profiles so far.